### ISSUE TITLE
Model overfitting | Text classification

### ISSUE THREAD

**Issue by usmaann** [amaiya/ktrain#219]
Hi 

Need some guidance.

I am training one model. I am getting good training and validation accuracy (98%) but when I test the model on my test data then accuracy drops a lot. it drops to 80%.

Any suggestion on how to improve performance on test data?

PS: My data is text (tweets only). and m only fine-tuning a pre-trained model on my data. and my using Ktrain library with Bert model

thanks.

**Comment by amaiya** [comment]
[This stackoverflow post](https://stackoverflow.com/questions/48718663/validation-and-testing-accuracy-widely-different) does a good job of various reasons why test accuracy will be lower than validation accuracy.  The two most likely reasons are:
1. sample sizes for validation and test are small so performance may vary a lot across different samples 
2. hyperparameter overfitting:  you see this happening in  Kaggle competitions

To shed more light on things, you can try switching the validation and test datasets.  If you see that the validation accuracy drops to 80% and the test accuracy increases significantly, then the cause is probably No. 1 in which case you can collect a larger sample for validation and test to get a more accurate picture of current performance.  Btw, this is the whole motivation behind cross-validation (which you can try to do but may take a long time with a model like BERT).

**Comment by usmaann** [comment]
Hi..Thanks.
.. I used a weight decay parameter given ktrain to avoid overfitting.
I used (0.01). and it slightly improved the performance on test data..score jumped from 80 to 81.25%... what do u suggest? should I try different parameter for weight decay now? What it should be?

also, I used Strathfield kfold (10) for training....
